test/test_images/event-flaker/main.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/kelseyhightower/envconfig"
+	_ "knative.dev/pkg/system/testing"
+)
+
+type envConfig struct {
+	Reject int    `envconfig:"REJECT_MOD" default:"3"`
+	Sink   string `envconfig:"K_SINK" required:"true"`
+}
+
+func main() {
```

Comment:
Why we need that? `eventshub` can do that and report when events are rejected, simplifying debugging. Let's avoid the proliferation of test images

Reply:
I think there isn't any option here, but you can add it: https://github.com/knative-sandbox/reconciler-test/blob/main/pkg/eventshub/options.go

eventshub can be configured for dropping events following a precise pattern: 

* https://github.com/knative-sandbox/reconciler-test/blob/main/pkg/test_images/eventshub/receiver/receiver.go#L76
* https://github.com/knative-sandbox/reconciler-test/blob/main/pkg/test_images/eventshub/receiver/receiver.go#L80

Reply:
Could we land this PR, and then work a followup to remove this and show me how to use events hub to fail 2 times and then accept an event?

Reply:
If I need to change events up then I think we merge this and fix up events up with this feature.